src/metabase/models/field_values.clj

```diff
@@ -5,17 +5,23 @@
              [db :as db]
              [models :as models]]))
 
-(def ^:const ^Integer low-cardinality-threshold
-  "Fields with less than this many distinct values should automatically be given a special type of `:type/Category`."
-  300)
+(def ^:const ^Integer category-cardinality-threshold
+  "Fields with less than this many distinct values should automatically be given a special type of `:type/Category`, and
+  a `has_field_values` value of `search`. This means we should cache the FieldValues "
+  (int 30))
+
+(def ^:const ^Integer list-cardinality-threshold
+  "Fields with less than this many distincy values should be given a `has_field_values` value of `list`, which means we
+  should do ad-hoc queries to fetch values of this Field for search widget purposes."
+  (int 100))
 
 (def ^:private ^:const ^Integer entry-max-length
   "The maximum character length for a stored `FieldValues` entry."
-  100)
+  (int 100))
```

Comment:
Where is this length used?

It would be nice if a user can override our classifier and treat say a 110 cardinality column as a "list". This is just our first, best guess on startup, rather than a hard+fast rule for all time.

Reply:
this isn't cardinality it's the maximum *character length* for a single entry.